spec/schema_spec.rb

```diff
@@ -60,12 +60,20 @@
   describe 'Array of Hashes' do
     let(:value) { [ { id: 1, contents: 'some text' }, { id: 2 } ] }
     it 'is an array containing the schema' do
-      expect(schema).to match(hash_including(
-          properties: [ 
-            { name: :id, class: 'Integer' },
-            { name: :contents, class: 'String' }
+      expect(schema).to match(
+        hash_including(
+          class: 'Array',
+          properties: [
```

Comment:
What does `class: Array` add, given that `properties` is already known to be an Array?

Reply:
Let me double check that I'm not emitting redundant type information before I respond to this.

Reply:
Here's an example of the current state:
```json
{
  "return_value": {
    "class": "Array",
    "value": "[{mapset_id=>13, scenario_uuid=>03aee8be-b5a0-494a-8a87-a1732bc7dd29...",
    "object_id": 95540,
    "size": 20,
    "properties": [
      { "name": "mapset_id", "class": "Integer" },
      { "name": "scenario_uuid", "class": "String" },
      { "name": "code_objects", "class": "Array" },
      {
        "name": "metadata",
        "class": "Hash",
        "properties": [
          { "name": "app", "class": "String" },
          { "name": "git", "class": "Hash" },
          { "name": "name", "class": "String" },
          { "name": "client", "class": "Hash" },
          { "name": "language", "class": "Hash" },
          { "name": "recorder", "class": "Hash" },
          { "name": "timestamp", "class": "Float" },
          { "name": "frameworks", "class": "Array" }
        ]
      },
      { "name": "labels", "class": "Array" }
    ]
  }
}
```

The `return_value` is an array and `properties` describes the schema of the object contained within. The above schema implicitly implies everything within the array is an object. We'll miss primitives, mixed types or nested arrays.

Here's an example of an API which returns the JSON response `[1, 2, 3]`:
```json
{
  "return_value": { "class": "Array", "value": "[1, 2, 3]", "object_id": 99920, "size": 3 },
}
```

We lose the type information because we're not capturing it. This is why `class: Array` is specified at this level. To compare, here are the previous outputs after the changes in this PR:

```json
{
  "return_value": {
    "class": "Array",
    "value": "[{mapset_id=>13, scenario_uuid=>03aee8be-b5a0-494a-8a87-a1732bc7dd29...",
    "object_id": 378440,
    "size": 20,
    "items": [
      {
        "class": "Hash",
        "properties": [
          { "name": "mapset_id", "class": "Integer" },
          { "name": "scenario_uuid", "class": "String" },
          { "name": "code_objects", "class": "Array", "items": [] },
          {
            "name": "metadata",
            "class": "Hash",
            "properties": [
              { "name": "app", "class": "String" },
              { "name": "git", "class": "Hash" },
              { "name": "name", "class": "String" },
              { "name": "client", "class": "Hash" },
              { "name": "language", "class": "Hash" },
              { "name": "recorder", "class": "Hash" },
              { "name": "timestamp", "class": "Float" },
              { "name": "frameworks", "class": "Array", "items": [{ "class": "Hash" }] }
            ]
          },
          { "name": "labels", "class": "Array", "items": [] }
        ]
      }
    ]
  }
}
```

```json
{
  "return_value": {
    "class": "Array",
    "value": "[1, 2, 3]",
    "object_id": 94980,
    "size": 3,
    "items": [{ "class": "Integer" }]
  }
}
```
It's more verbose but it's necessary for us to describe the element type of any array. It could be argued that any property object containing `properties` or `items` doesn't need `class` specified (in this case we know it's an `object` or `array`) but I'd err on the side of retaining more data.